### PR TITLE
Milestone 4

### PR DESCRIPTION
**Pull request by Israel-Laguan**
# Features

- Facebook-like frontend
image
image
image
- Like posts from Timeline
image
image
image
image

### CODE REVIEW

File: app/models/like.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# frozen_string_literal: true
+
+class Like < ApplicationRecord
+  belongs_to :likeable, polymorphic: true
+  has_one :user
```
Comment (samgaco) [inline app/models/like.rb]: - [ ] each like should also belong to one post

**Review by samgaco — CHANGES_REQUESTED** [review]
Hi @Israel-Laguan and @raheebwa :high_brightness: 

- [x] For this milestone, and user [should be able to like](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone4/likes.png) posts, [comment](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone4/usr_comments.png) on posts and the posts should always be [displayed with its comments.](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone3/posts4.png)

> notable


------------------

- [x] Also, have into account that the signup form now looks like this (the name, last name, location fields appear as password fields):
problem

And if I try to create and user from there I get the following:
error

-------------------
Moving on :arrow_down: 

* **[For your next code review requests]**  Use meaningful names for your Pull Requests
* **[For your next code review requests]**  Add a short summary of changes in Pull Request description


-------------------


Please, work on these changes and submit back for another code review once you are done :muscle: 


* ###### STATUS OF THE PROJECT: REQUESTED CHANGES :stop_sign:

File: app/models/like.rb (hunk 1 above)
Comment (Israel-Laguan) [reply app/models/like.rb]: https://github.com/raheebwa/irla-facebook/blob/5b15c0d6476da84a7e7095a5647472b45c511428/app/models/like.rb#L3-L9

Solved!

**Comment by Israel-Laguan** [comment]
About your review:
> 
>     * [x]  For this milestone, and user [should be able to like](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone4/likes.png) posts, [comment](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone4/usr_comments.png) on posts and the posts should always be [displayed with its comments.](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone3/posts4.png)

- [x] Users can like posts
image
image

- [x] Users can like comments
image

image

- [x] users can comment

- [x] posts display comments when you inspect the post. We decided that way because showing all the comments on Timeline would increase the complexity. 

> 
>     * [x]  Also, have into account that the signup form now looks like this (the name, last name, location fields appear as password fields):
>       problem
> 
 Solved!: 
image

> 
> And if I try to create and user from there I get the following:
> error
> 
Solved!!

Thanks for your review!

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -61,6 +61,31 @@ def destroy
...
+    current_user.likes.build(post_id: @post.id)
+    if current_user.save
+      redirect_back fallback_location: root_path, notice: 'You liked a comment!.'
+    else
+      redirect_to root_path, alert: 'You already liked this post'
+    end
+  end
+
+  def like_comment
+    @post.likes.build(user_id: params[:user_id], comment_id: params[:comment_id]).save
```
Comment (efrapp) [inline app/controllers/posts_controller.rb]: You get an error at this line, please see #25

**Review by efrapp — CHANGES_REQUESTED** [review]
## Hi @Israel-Laguan and @raheebwa 👋
Nice job guys, you met all the checkpoints from the previous code review, however, when I was checking the app I caught up an error that you need to address. Please see my comments below:

### Last review summarize
- [X] User should be able to like posts
- [X] User should be able to comment on posts
- [X] The post is always displayed with its comments
- [X] Fix name, last name, and location fields
- [X] User meaningful name in the PR
- [X] Add short summarize description to the PR

### Required changes
- [ ] Fix the bug, see #25
- [ ] You have a failing test, see #26

As a suggestion, in this image:
Screen Shot 2019-12-10 at 4 27 43 PM
It isn't clear for me what is the purpose of those like buttons in the comments. They are giving the impression that you can like comments but if that is the case, you need to fix the logic because each time a user likes a post all the likes on comments are updated to the same amount of likes the post has so it is a bit confusing.

Please, send a new code review when you fix the bug.

Have a great day!

**Status**: `Changes required`

**Review by adriaanbd — APPROVED** [review]
This PR now meets all required specifications for this milestone. You may now merge.

Status: `Approved`

**Pull request merged**